### DISCUSSION TITLE
How to access the nested year property of a datetime column?

### DISCUSSION THREAD

**Discussion by Josrph** [prisma/prisma#17247] [Q&A]
suppose I have a model like so:

```
model User {
  id      Int      @id @default(autoincrement())
  name    String
  createdAt    DateTime
}
```

how to access the nested year property of the createdAt column?

**Comment by nurul3101** [comment]
Hello @Josrph 👋 
You can use [@db.year](https://www.prisma.io/docs/reference/api-reference/prisma-schema-reference#mysql-2) attribute in case you are using MySQL and just want to store the year value. Else, you would need to get the year value after the record is fetched.

PrismaClient returns DateTime as native JavaScript date objects. You can get the year value like this:
```javascript
let dateString = "2023-01-01T00:00:00.000Z";
let date = new Date(dateString);
let year = date.getFullYear();
console.log(year) // 2023
```

**Reply by Josrph** [reply]
Thanks for your reply, but what I want is to query the year before getting the data and not after.

**Reply by nurul3101** [reply]
I am not sure if I understand correctly, but to get the year it would be necessary to get the data first.

**Reply by Josrph** [reply]
What you're proposing is not efficient because first you need to get all the records from the database and then filter the collection in the server, But what I want is to get the data filtered directly from the database.

for example: get all records of the year 2020 only and nothing else with it, directly from the database.

**Reply by nurul3101** [reply]
Oh I see what you are trying to achieve. Can you try using the [lt](https://www.prisma.io/docs/reference/api-reference/prisma-client-reference#lt) and [gt](https://www.prisma.io/docs/reference/api-reference/prisma-client-reference#gt) operators to get all the records that lie from 01-01-2020 to 31-12-2022? Here's an [example](https://www.prisma.io/docs/reference/api-reference/prisma-client-reference#get-all-post-records-where-date_created-is-greater-than-march-19th-2020) of what I am suggesting.

**Comment by ludralph** [comment]
Hi there,

To keep our discussions organized and focused on the most relevant topics, we’re reviewing and tidying up our backlog. As part of this process, we’re closing discussions that haven’t had any recent activity and appear to be outdated.

If this discussion is still important to you or unresolved, we’d love to hear from you! Feel free to reopen it or start a new one with updated details.

For more details about our priorities and vision for the future of Prisma ORM, check out our latest blog post: https://www.prisma.io/blog/prisma-orm-manifesto.

Thank you for your understanding and being part of the community!
